staging/src/k8s.io/kubectl/pkg/drain/cordon_test.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package drain
+
+import (
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type newCordonHelperFromRuntimeObjectTestCase struct {
+	name        string
+	nodeObject  runtime.Object
+	expectError bool
+	expected    *CordonHelper
+}
+
+func TestNewCordonHelperFromRuntimeObject(t *testing.T) {
+	tests := []newCordonHelperFromRuntimeObjectTestCase{
+		{
+			name: "valid node object",
+			nodeObject: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-node",
+				},
+			},
+			expectError: false,
+			expected: &CordonHelper{
+				node: &corev1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-node",
+					},
+				},
+			},
+		},
+		{
+			name: "invalid object type",
+			nodeObject: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-pod",
+				},
+			},
+			expectError: true,
+			expected:    nil,
+		},
+	}
+	scheme := runtime.NewScheme()
+	_ = corev1.AddToScheme(scheme)
+	gvk := schema.GroupVersionKind{
+		Group:   "",
+		Version: "v1",
+		Kind:    "Node",
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			helper, err := NewCordonHelperFromRuntimeObject(tt.nodeObject, scheme, gvk)
+			if tt.expectError && err == nil {
+				t.Error("Expected error but got none")
+			}
+			if !tt.expectError && err != nil {
+				t.Errorf("Unexpected error: %v", err)
+			}
+			if !tt.expectError && helper == nil {
+				t.Error("Expected non-nil helper")
+			}
+			if tt.expected != nil && helper != nil {
+				if helper.node.Name != tt.expected.node.Name {
```

Comment:
I don't think, comparing the node name only would suffice. What about cmp.Diff, etc.?

Reply:
You're absolutely right (it wasn't sufficient, as you can see, lol). I addressed it in [f544584](https://github.com/kubernetes/kubernetes/pull/129202/commits/f5445844d218e98f73a11b82dcae7435f09002ea).